### ISSUE TITLE
MessageInputBar overlaps navigationBar in landscape orientation

### ISSUE THREAD

**Issue by nasserprofessional** [MessageKit/MessageKit#345]
## General Information

* MessageKit Version: 0.10.1

* iOS Version(s): 11.0.3

* Swift Version: 4

* Devices/Simulators: iPhone 7 and simulator

* Reproducible in ChatExample? (Yes/No): Yes

##  What happened?

    1. Launch Example project in portrait
    2. Start typing and press return on the keyboard to create multiple lines
    3. Rotate to landscape
The MessageInputBar overlaps the navigationBar

## What did you expect to happen?
I expected the MessageInputBar to stop growing when it reaches the navigationBar

![32680108-17a8903a-c661-11e7-951e-9144b1b439c4](https://user-images.githubusercontent.com/476105/32690866-e3ae154e-c6f5-11e7-9581-24e417c324e2.png)

**Comment by nasserprofessional** [comment]
I have managed to fix the issue in the Example but Im not sure if this is the best way to do it:
```swift
 override func viewDidLoad() {
        super.viewDidLoad()
        
        
        NotificationCenter.default.addObserver(
            self,
            selector: #selector(keyboardChangedFrame),
            name: NSNotification.Name.UIKeyboardDidChangeFrame,
            object: nil
        )

}


@objc func keyboardChangedFrame(_ notification: Notification) {
        if let keyboardFrame: NSValue = notification.userInfo?[UIKeyboardFrameEndUserInfoKey] as? NSValue {

//getting the keyboard height
            let keyboardRectangle = keyboardFrame.cgRectValue
            let keyboardHeight = keyboardRectangle.height
            
            
            //getting the navigationBar height
            let navBarHeight = UIApplication.shared.statusBarFrame.height +
                self.navigationController!.navigationBar.frame.height
         
            /*setting the messageInputBar.maxHeight to the area above the keyboard:*/

            messageInputBar.maxHeight = UIScreen.main.bounds.size.height - keyboardHeight - navBarHeight + self.messageInputBar.frame.size.height - 16

/*16 seems to be the insets of 8 that are at the top and bottom of the textview I believe we have to subtract these because we are setting the height constraint on the textView and not the container so we need to take into account any spacing around it. Theres probably a much better way of doing this by setting the height constraint on the container directly.*/
            
            
        }
    }
```

![614fc520-febf-478f-873c-56f9beecdb30](https://user-images.githubusercontent.com/476105/32690907-04f1c1c8-c6f7-11e7-8148-cec6eb7719b7.png)

**Comment by nathantannar4** [comment]
Hi @nasserprofessional 

To allow of for flexibility we made it so that the `maxHeight` is an easy to set variable that will auto adjust the view. We may not want to add an explicit set of the `maxHeight` during keyboard frame change as it reduces the flexibility. If you want to make a PR for this adding this auto set option I encourage you to make one for the `MessagesViewController` with a boolean like `autoAdjustMaxMessageInputBarHeight`

**Comment by nathantannar4** [comment]
Note: Possible solution may be adding a topAnchor constraint to the top of the window. Looking into possibility

**Comment by martinpucik** [comment]
This should be resolved in #1704
